src/blocksigner.cpp

```diff
@@ -69,8 +73,13 @@ bool CBlockSigner::SignBlock()
             scriptType = GetScriptTypeFromDestination(destination);
         }
     }
-//?
-    return CHashSigner::SignHash(refBlock.IsTPoSBlock() ? refBlock.GetTPoSHash() : refBlock.GetHash(), keySecret, scriptType, refBlock.vchBlockSig);
+
+    const auto &hash = refBlock.IsTPoSBlock() ? refBlock.GetTPoSHash() : refBlock.GetHash();
+    if (nChainHeight >= Params().GetConsensus().nTPoSSignatureUpgradeHFHeight) {
```

Comment:
I'd suggest to create a common function that has tests to check this, instead of keep writing the same check everywhere.

Or, even better, just write a helper thing that handles signing contracts, and verifying signatures.